### PR TITLE
OD-16387 New columns for all list views

### PR DESCRIPTION
**Pull request by dmitars**
…ith bad performance

### CODE REVIEW

File: server/src/main/groovy/ish/oncourse/server/preference/DefaultUserPreference.groovy [hunk 1]
```diff
@@ -843,41 +909,41 @@ class DefaultUserPreference {
 
     // must be below model initializations
     public static final Map<String, TableModelDTO> DEFAULT_MODEL_MAP = [
```
Comment (ArtyomKravchenko) [inline server/src/main/groovy/ish/oncourse/server/preference/DefaultUserPreference.groovy:911]: Please do not make a changes with no reason, it make harder to review

File: types/src/main/java/ish/util/InvoiceUtil.java [hunk 2]
```diff
@@ -395,6 +395,24 @@ public static Money amountOwingForPayer(ContactInterface payer) {
 		return result;
 	}
 
+	/**
+	 * sums all invoice amount overdue for a payer
+	 *
+	 * @param payer to be analysed
+	 * @return Money sum of amount overdue
+	 */
+	public static Money amountOverdueForPayer(ContactInterface payer) {
```
Comment (ArtyomKravchenko) [inline types/src/main/java/ish/util/InvoiceUtil.java:404]: ContactInterface is wrong type here, why do not you use just Contact?

Comment (dmitars) [reply types/src/main/java/ish/util/InvoiceUtil.java:404]: Contact has a package-frendly access modifier, and i think, to use interface here is more safe than make modifier public. Now class hasn't hard dependencies from entity types and belongs to the next level of architecture

File: server-api/src/main/groovy/ish/oncourse/server/api/traits/OutcomeProgressionDTOTrait.groovy [hunk 3]
```diff
@@ -1,5 +1,35 @@
 package ish.oncourse.server.api.traits
 
 trait OutcomeProgressionDTOTrait {
+    abstract BigDecimal getAbsent();
 
+    abstract BigDecimal getAttended();
+
+    abstract BigDecimal getNotMarked();
```
[the remark this answers is not in the document]
Comment (dmitars) [reply server-api/src/main/groovy/ish/oncourse/server/api/traits/OutcomeProgressionDTOTrait.groovy:8]: It is a necessity. These methods are needed to make calculations, but their implementation is automatically generated in OutcomeProgressionDTO class

**Pull request closed without merging**